analyses/infercnv-consensus-cell-type/exploratory-notebooks/01_ewings-consensus-cell-types.Rmd

```diff
@@ -0,0 +1,188 @@
+```
+
+`inferCNV` authors [recommend that at least 100 cells](https://github.com/broadinstitute/infercnv/issues/188) be used for a "normal" reference, so we are above that threshold.
+
+If we were to use all of these, what fraction is that per sample?
+We'll just divide the total number of cells by this count, not taking into consideration how many immune cells are actually in that sample, to get a rough sense of how many immune cells there are relative to all.
+
+```{r}
+celltype_df |>
+  dplyr::count(library_id, name = "total_cells") |>
+  dplyr::mutate(fraction_immune = round(total_immune / total_cells, 2)) |>
+  knitr::kable()
+```
+
+For the low-quality sample `SCPCL001111` this fraction isn't very meaningful since there are fewer cells in the sample than there are pooled immune cells.
+For all other samples, the fraction ranges from 0.03 - 0.13.
+
+
+Let's look a bit more at the breakdown of immune cell types:
+
+```{r}
+celltype_df |>
+  dplyr::filter(category == "immune") |>
+  dplyr::count(library_id, consensus_annotation) |>
+  # this will arrange columns (after pivoting) in descending order of total counts
+  dplyr::arrange(desc(n)) |>
+  tidyr::pivot_wider(
+    names_from = consensus_annotation,
+    values_from = n,
+    values_fill = 0
+  )
+```
+
+
+Let's look at the total number of macrophages and all types of T cell, since these are the dominant cell types here.
+
+```{r}
+celltype_df |>
+  dplyr::mutate(
+    consensus_annotation = ifelse(
+      stringr::str_detect(consensus_annotation, "T cell"),
+      "T cell type",
+      consensus_annotation
+    )
+  ) |>
+  dplyr::filter(consensus_annotation %in% c("T cell type", "macrophage")) |>
+  dplyr::count(consensus_annotation)
+```
+These constitute the vast majority of cells, so it would also be an option to use just these groups of cell types.
+If we were to want a single immune cell type, then macrophage is the only suitable option to get more than 100 cells.
+
+
+## Compare to ewings cell typing
+
+We should also check whether any of the immune cells were classified as a tumor cell in the `cell-type-ewings` analysis module.
+Ideally this will not have been the case, but it's an important thing to check!
+
+```{r}
+celltype_df |>
+  dplyr::filter(category == "immune") |>
+  dplyr::count(ewing_annotation) |>
+  dplyr::filter(stringr::str_detect(ewing_annotation, "tumor"))
+```
+
+Well, it appears there are some consensus immune cells which were classified as `tumor`. 
+Which samples are these in?
+
+```{r}
+celltype_df |>
+  dplyr::filter(
+    stringr::str_detect(ewing_annotation, "tumor"),
+    category == "immune"
+  ) |>
+  dplyr::count(library_id, ewing_annotation, consensus_annotation)
+```
+
+These results show macrophage and T cells, which are the two most prevalent categories immune cell categories.
+We might want to investigate these cells further, but at least we should probably exclude them as "normal" cells for `inferCNV` since their true status is bit unclear.
```

Comment:
```suggestion
We might want to investigate these cells further, but at least we should probably exclude them as "normal" cells for `inferCNV` since their true status is a bit unclear.
```